File: C++/Program/ParkingLot.cpp
/* 
    Saya Adrian Mulianto NIM [2200939] mengerjakan soal Latihan Praktikum-2 dalam mata kuliah DPBO
    untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin 
*/

// Menggunakan pragma once agar header file hanya di-include satu kali
#pragma once
// Memasukan library yang digunakan
#include <iostream>
#include <string>

// Deklarasi namespace
using namespace std;

/*
    Kelas ParkingLot yang digunakan untuk merepresentasikan data ParkingLot
*/
class ParkingLot {
private:
    // Atribut yang digunakan pada kelas ParkingLot terdiri dari merk ParkingLot beserta jenis prosesornya
    string kapasitas;
    string jumlah_kendaraan_saat_ini;
public:
    // Constructor
    ParkingLot(){

    }

    // Constructor : Overloading
    ParkingLot(string kapasitas){
        this->kapasitas = kapasitas;
    }

    // Constructor : Overloading
    ParkingLot(string kapasitas, string jumlah_kendaraan_saat_ini){
        this->kapasitas = kapasitas;
        this->jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini;
    }

    // Method setter untuk set nilai setiap atribut pada kelas ParkingLot
    void setkapasitas(string kapasitas){
        this->kapasitas = kapasitas;
    }

    void setjumlah_kendaraan_saat_ini(string jumlah_kendaraan_saat_ini){
        this->jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini;
    }

    // Method getter untuk mendapatkan nilai setiap atribut pada kelas ParkingLot
    string getkapasitas(){
        return kapasitas;
    }

    string getjumlah_kendaraan_saat_ini(){
        return jumlah_kendaraan_saat_ini;
    }

    // Destructor
    ~ParkingLot(){

    }
};

File: Python/Program/ParkingLot.py

    # Saya Adrian Mulianto NIM [2200939] mengerjakan soal Latihan Praktikum-4 dalam mata kuliah DPBO
    # untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin 

class ParkingLot:
    def __init__(self, kapasitas = "", jumlah_kendaraan_saat_ini = ""):
        self.kapasitas = kapasitas
        self.jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini

    # Method setter untuk set nilai setiap atribut pada kelas ParkingLot
    def set_kapasitas(self, kapasitas):
        self.kapasitas = kapasitas

    def set_jumlah_kendaraan_saat_ini(self, jumlah_kendaraan_saat_ini):
        self.jumlah_kendaraan_saat_ini = jumlah_kendaraan_saat_ini

    # Method getter untuk mendapatkan nilai setiap atribut pada kelas ParkingLot
    def get_kapasitas(self):
        return self.kapasitas

    def get_jumlah_kendaraan_saat_ini(self):
        return self.jumlah_kendaraan_saat_ini
